Frameworks/UIKit/UICGColor.mm

```diff
@@ -0,0 +1,270 @@
+        case 3:
+            out.r = p;
+            out.g = q;
+            out.b = in.v;
+            break;
+        case 4:
+            out.r = t;
+            out.g = p;
+            out.b = in.v;
+            break;
+        case 5:
+        default:
+            out.r = in.v;
+            out.g = p;
+            out.b = q;
+            break;
+    }
+    return out;
+}
+
+#pragma endregion helpers
+
+#pragma region Inits
+
+@implementation _UIColorConcrete {
+    woc::StrongCF<CGColorRef> _cgColor;
+}
+
+- (instancetype)initWithHue:(CGFloat)h saturation:(CGFloat)s brightness:(CGFloat)v alpha:(CGFloat)a {
+    hsv in;
+    rgb out;
+
+    in.h = h * 360.0f;
+    in.s = s;
+    in.v = v;
+
+    out = hsv2rgb(in);
+
+    return [self initWithRed:static_cast<float>(out.r) green:static_cast<float>(out.g) blue:static_cast<float>(out.b) alpha:a];
+}
+
+- (instancetype)initWithRed:(CGFloat)r green:(CGFloat)g blue:(CGFloat)b alpha:(CGFloat)a {
+    if (self = [super init]) {
+        _cgColor.attach(CGColorCreateGenericRGB(r, g, b, a));
+    }
+
+    return self;
+}
+
+- (instancetype)initWithWhite:(CGFloat)gray alpha:(CGFloat)alpha {
+    if (self = [super init]) {
+        _cgColor.attach(CGColorCreateGenericGray(gray, alpha));
+    }
+
+    return self;
+}
+
+- (instancetype)initWithPatternImage:(UIImage*)image {
+    CGImageRef img = static_cast<CGImageRef>([image CGImage]);
+    if (img == nullptr) {
+        [self release];
+        return nil;
+    }
+
+    if (self = [super init]) {
+        auto pattern = woc::MakeStrongCF<CGPatternRef>(_CGPatternCreateFromImage(img));
+        _cgColor.attach(CGColorCreateWithPattern(CGImageGetColorSpace(img), pattern, nullptr));
+    }
+
+    return self;
+}
+
+- (instancetype)initWithCGColor:(CGColorRef)cgColor {
+    if (cgColor == nullptr) {
+        [self release];
+        return nil;
+    }
+
+    if (self = [super init]) {
+        _cgColor = cgColor;
+    }
+
+    return self;
+}
+
+#pragma endregion Inits
+
+- (void)set {
+    //  Set current context color
+    if (UIGraphicsGetCurrentContext()) {
+        CGContextSetFillColorWithColor(UIGraphicsGetCurrentContext(), _cgColor);
+        CGContextSetStrokeColorWithColor(UIGraphicsGetCurrentContext(), _cgColor);
+    } else {
+        TraceVerbose(TAG, L"UIColor::set - context not set");
+    }
+}
+
+- (void)setFill {
+    //  Set current context color
+    if (UIGraphicsGetCurrentContext()) {
+        CGContextSetFillColorWithColor(UIGraphicsGetCurrentContext(), _cgColor);
+    } else {
+        TraceVerbose(TAG, L"UIColor::setFill - context not set");
+    }
+}
+
+- (void)setStroke {
+    //  Set current context color
+    CGContextSetStrokeColorWithColor(UIGraphicsGetCurrentContext(), _cgColor);
+}
+
+- (UIColor*)colorWithAlphaComponent:(CGFloat)alpha {
+    auto color = woc::MakeStrongCF<CGColorRef>(CGColorCreateCopyWithAlpha(_cgColor, alpha));
+    return [UIColor colorWithCGColor:color];
+}
+
++ (UIColor*)colorWithColor:(UIColor*)copyclr {
```

Comment:
this doesn't seem to be used anywhere, and isn't a part of the docs. let's get rid of it (if we do need it for some reason, it seems like [[copyclr copy] autorelease] would be the more sensical return value?